Lib/test/test_email/test_headerregistry.py

```diff
@@ -1437,6 +1437,18 @@ def test_il8n(self):
     #    with self.assertRaises(ValueError):
     #        Address('foo', 'wők', 'example.com')
 
+    def test_crlf_in_display_name_raises(self):
+        with self.assertRaisesRegex(ValueError, "invalid address"):
+            Address(display_name='example.com\r\n')
+
+    def test_crlf_in_username_raises(self):
+        with self.assertRaisesRegex(ValueError, "invalid address"):
+            Address(username='hello\r\n')
+
+    def test_crlf_in_domain_raises(self):
+        with self.assertRaisesRegex(ValueError, "invalid address"):
+            Address(domain='example.com\r\n')
```

Comment:
Need a test for crlf in addr_spec as well, and for just CR and just LF in each one.  You could use the email package's test parameterization facility to do this with fewer lines of code, but that's not required :)

Reply:
Done.

Reply:
I ended up using subTest because it seemed cleaner.